pkg/odo/cli/component/common_push.go

```diff
@@ -330,3 +332,71 @@ func filterIgnores(filesChanged, filesDeleted, absIgnoreRules []string) (filesCh
 	}
 	return filesChangedFiltered, filesDeletedFiltered
 }
+
+// retrieveDefaultComponentNamespace tries to retrieve the current active namespace
+// to set as a default namespace
+func retrieveKubernetesDefaultNamespace() (string, error) {
+	// Get current active namespace
+	client, err := kclient.New()
+	if err != nil {
+		return "", err
+	}
+	return client.Namespace, nil
+}
+
+// checkNamespaceDefaultIfNotAvailable checks to see if a namespace has been passed in, if not, we will default
+// to the standard kubernetes namespace.
+func checkCmdNamespaceDefaultIfNotAvailable(cmd *cobra.Command) (string, error) {
+	var componentNamespace string
+
+	// For "odo create" check to see if --project has been passed.
+	if cmd.Flags().Changed("project") {
+		componentNamespace, err := cmd.Flags().GetString("project")
+		if err != nil {
+			return componentNamespace, err
+		}
+	} else if cmd.Flags().Changed("namespace") {
+		// For "odo push" check to see if project has been passed
+		componentNamespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			return componentNamespace, err
+		}
+	}
+
+	componentNamespace, err := retrieveKubernetesDefaultNamespace()
+	if err != nil {
+		return "", err
+	}
+
+	return componentNamespace, nil
+}
+
+// gatherName parses the Devfile retrieves an appropriate name in two ways.
+// 1. If metadata.name exists, we use it
+// 2. If metadata.name does NOT exist, we use the folder name where the devfile.yaml is located
+func gatherName(devfilePath string) (string, error) {
+
+	// Parse the Devfile
+	devObj, err := devfile.Parse(devfilePath)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to parse devfile")
```

Comment:
Should we make it `unable to parse devfile while trying to get component name` or something like that? Asking because we're parsing devfile at https://github.com/openshift/odo/blob/b815649d4f207628c5ed91bbcaafea9358d16ffb/pkg/odo/cli/component/create.go#L724-L727
as well. And I won't be surprised if more calls get added in future. Just that it would be helpful to understand where exactly we fail to parse a devfile.

Quite likely a nitpick. :wink: